Sources/GitHub/API.swift

```diff
@@ -2,64 +2,144 @@ import Foundation
 import Sugar
 
 open class API {
-	private let accessToken: String
+	typealias HTTPHeader = (field: String, value: String?)
 
-	public init(accessToken: String) {
-		self.accessToken = accessToken
+	let authorization: Authorization
+
+	init(authorization: Authorization) {
+		self.authorization = authorization
+	}
+
+	func get<ResponseModel: Codable>(
+		endpoint: String?,
+		queryItems: [URLQueryItem]? = nil,
+		headers: [HTTPHeader] = [],
+		responseType: ResponseModel.Type
+	) throws -> ResponseModel {
+		return try perform("GET", endpoint: endpoint, queryItems: queryItems, headers: headers, responseType: responseType)
 	}
 
-	func get<EndpointModel: Codable>(
-		_ model: EndpointModel.Type,
-		for shorthand: Repository.Shorthand,
+	func post<RequestModel: Encodable, ResponseModel: Decodable>(
 		endpoint: String?,
-		queryItems: [URLQueryItem]? = nil
-	) throws -> EndpointModel {
-		let endpoint = endpoint.map { "/\($0)" } ?? ""
+		queryItems: [URLQueryItem]? = nil,
+		headers: [HTTPHeader] = [],
+		body: RequestModel,
+		responseType: ResponseModel.Type
+	) throws -> ResponseModel {
+		return try perform("POST", endpoint: endpoint, queryItems: queryItems, headers: headers, body: JSONEncoder().encode(body), responseType: responseType)
+	}
+
+	func patch<RequestModel: Encodable, ResponseModel: Decodable>(
```

Comment:
<!--
  1 failure:  Prefer to locate ...
  0 warning: 
  
  
  DangerID: danger-id-default;
  File: Sources/GitHub/API.swift;
  Line: 32;
-->
- :no_entry_sign: Prefer to locate parameters with defaults toward the end of the parameter list. (`function_default_parameter_at_end`)